### DISCUSSION TITLE
How to create random circles in a specific region

### DISCUSSION THREAD

**Discussion by patte103** [idaholab/moose#20216] [Q&A Modules: Phase field] [answered]
Hello!
I am currently using "MultiSmoothCircleIC" to create circles in my generated mesh. Once I increase the size of my mesh domain and create a smaller subdomain with "SubdomainBoundingBoxGenerator" where I want the circles to be generated, the "MultiSmoothCircleIC" keeps generating the circles outside of the subdomain. Is there a way to generate random circles within a specified region? Or is it better to just use "SpecifiedSmoothCircleIC"?

Thanks!

**Comment by SudiptaBiswas** [comment, marked as answer]
You have to block restrict your initial condition to generate circles within the subdomain. Try specifying `block=#subdomain_id` in your initial condition.
